test/functional/mempool_accept.py

```diff
@@ -58,38 +55,31 @@ def check_mempool_result(self, result_expected, *args, **kwargs):
 
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
 
-        self.log.info('Start with empty mempool, and 200 blocks')
+        self.log.info('Start with empty mempool, and 400 blocks')
         self.mempool_size = 0
-        assert_equal(node.getblockcount(), 200)
+        miniwallet.generate(200)
```

Comment:
Generating blocks takes a few seconds in valgrind, so I am thinking if this test may benefit from using `miniwallet.scan_blocks` (to be introduced in #21200)